daft/runners/native_runner.py

```diff
@@ -80,23 +81,35 @@ def run_iter(
         track_runner_on_scarf(runner=self.name)
 
         # NOTE: Freeze and use this same execution config for the entire execution
-        daft_execution_config = get_context().daft_execution_config
+        ctx = get_context()
+        query_id = str(uuid.uuid4())
 
         # Optimize the logical plan.
+        ctx.notify_query_start(query_id, repr(builder))
+        ctx.notify_plan_start(query_id)
         builder = builder.optimize()
+        ctx.notify_plan_end(query_id, repr(builder))
 
         # NOTE: ENABLE FOR DAFT-PROTO TESTING
         # builder = _to_from_proto(builder)
 
         plan = LocalPhysicalPlan.from_logical_plan_builder(builder._builder)
         executor = NativeExecutor()
+        ctx.notify_exec_start(query_id)
```

Comment:
I'm not really sure if this is the right place to put it. Maybe it makes sense to put it with the RuntimeSubscribers in Swordfish? They have an init and finalize there which are currently unused.

Plus, right now I have no way to actually get the LocalPipelineNodes. I have no need for the `LocalPhysicalPlan` anyways. Thoughts?